robot/impl/local_robot.go

```diff
@@ -690,7 +690,7 @@ func (r *localRobot) updateWeakDependents(ctx context.Context) {
 				r.Logger().Errorw("failed to reconfigure internal service", "service", resName, "error", err)
 			}
 		case framesystem.InternalServiceName:
-			fsCfg, err := r.FrameSystemConfig(ctx)
+			fsCfg, err := r.buildFrameSystemConfig(ctx)
```

Comment:
To clarify, you don't want to use cached parts here?

Reply:
I might be misunderstanding something here, but I think during an update to weak dependence we would want to update what is being cached in the frame system.  This caching is useful for when we are looking to consume the framesystem

Reply:
Cool, that sounds correct to me.